src/resmom/linux/alps.c

```diff
@@ -4790,9 +5556,22 @@ alps_create_reservation(basil_request_reserve_t *bresvp, long *rsvn_id,
 				param->nppn);
 			add_alps_req(utilBuffer);
 		}
-		if (vnode_per_numa_node && (param->segments[0] != '\0')) {
-			snprintf(utilBuffer, sizeof(utilBuffer),
-				" " BASIL_ATR_SEGMENTS "=\"%s\"",
+		if (param->pstate > 0) {
+			sprintf(utilBuffer, " " BASIL_ATR_PSTATE "=\"%ld\"",
+				param->pstate);
+			add_alps_req(utilBuffer);
+		}
+		if (param->nppcu > 0) {
+			sprintf(utilBuffer, " " BASIL_ATR_NPPCU "=\"0\"");
+			add_alps_req(utilBuffer);
+		}
+		if (param->pgovernor[0] != '\0') {
+			sprintf(utilBuffer, " " BASIL_ATR_PGOVERNOR "=\"%s\"",
+				param->pgovernor);
+			add_alps_req(utilBuffer);
+		}
```

Comment:
This code seems outside the scope of using the new BASIL 1.3/1.4 interface to fill in information for resources PBS already has.  If you plan to use it for another feature later, please introduce this code later with the other feature.